rfcs/0148-taskcluster-worker-location.md

```diff
@@ -0,0 +1,48 @@
+# RFC 148 - TASKCLUSTER_WORKER_LOCATION
+* Comments: [#148](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/148)
+* Proposed by: @djmitche
+
+# Summary
+
+This proposal defines an environment variable, `TASKCLUSTER_WORKER_LOCATION`, containing this information in an easily-digested format.
+
+## Motivation
+
+Workers often need information about where they are running, for example to select nearby mirrors or other resources.
+
+Taskcluster does not currently provide this in a documented, supported fashion.
+
+# Details
+
+This proposes to introduce `$TASKCLUSTER_WORKER_LOCATION` as an environment variable available within tasks run by docker-worker and generic-worker.
+
+The content has the format `<cloud>/<details>`.
+The `<cloud>` portion will correspond to the provider types used within worker-manager (e.g., `google`, `amazon`).
+The `<details>` portion is specific to the cloud.
+Tasks not executed in any cloud (e.g., static or standalone workers) can define arbitrary locations.
+Examples:
+* `amazon/us-east-1b`
+* `google/us-east1-c`
```

Comment:
is there value in being more specific than 'region' in gcp/aws (as in the 1b/1c etc)

Reply:
In both cases it's easy to determine the region from the {availability-,}zone, and experience suggests that even if we can't imagine why, *someone* will want that information someday.  I have the vague sense that Google zones are more relevant to transit prices than Amazon AZ's, so it might make sense to have a per-zone mirror there.